utils/formUtils.ts

```diff
@@ -53,15 +53,17 @@ export function createStep({
 }
 
 export function setStepState(steps: Array<Step>, setSteps: Function, step: Step, state: any) {
-  const index = steps.findIndex((iStep) => step.section === iStep.section)
+  if (step.schemaRef !== steps[0].schemaRef) {
```

Comment:
I think we only want to keep on setting the step state when the schemas _do match up_, not when they don't.